draft-ietf-httpbis-message-signatures.md

```diff
@@ -275,74 +277,38 @@ The following table shows example canonicalized values for different content ide
 
 To differentiate specialty content identifiers from HTTP headers, specialty content identifiers MUST start with the "at" `@` character. This specification defines the following specialty content identifiers:
 
-@request-target
-: The target request endpoint. ({{content-request-target}})
-
 @signature-params
 : The signature metadata parameters for this signature. ({{signature-params}})
 
-Additional specialty content identifiers MAY be defined and registered in the HTTP Signatures Specialty Content Identifier Registry. ({{content-registry}})
+@method
+: The method used for a request. ({{content-request-method}})
 
-### Request Target {#content-request-target}
+@target-uri
+: The target URI for a request. ({{content-target-uri}})
 
-The request target endpoint, consisting of the request method and the path and query of the effective request URI, is identified by the `@request-target` identifier.
-
-Its value is canonicalized as follows:
-
-1. Take the lowercased HTTP method of the message.
-2. Append a space " ".
-3. Append the path and query of the request target of the message, formatted according to the rules defined for the :path pseudo-header in {{!HTTP2=RFC7540}}, Section 8.1.2.3.  The resulting string is the canonicalized value.
-
-#### Canonicalization Examples
-
-The following table contains non-normative example HTTP messages and their canonicalized `@request-target` values.
-
-<table>
-    <name>Non-normative examples of <tt>@request-target</tt> canonicalization.</name>
-    <thead>
-        <tr><th>HTTP Message</th><th>@request-target</th></tr>
-    </thead>
-    <tbody>
-        <tr>
-            <td><sourcecode type="http-message">
-POST /?param=value HTTP/1.1
-Host: www.example.com</sourcecode></td>
-            <td><tt>post /?param=value</tt></td>
-        </tr>
-        <tr>
-            <td><sourcecode type="http-message">
-POST /a/b HTTP/1.1
-Host: www.example.com</sourcecode></td>
-            <td><tt>post /a/b</tt></td>
-        </tr>
-        <tr>
-            <td><sourcecode type="http-message">
-GET http://www.example.com/a/ HTTP/1.1</sourcecode></td>
-            <td><tt>get /a/</tt></td>
-        </tr>
-        <tr>
-            <td><sourcecode type="http-message">
-GET http://www.example.com HTTP/1.1</sourcecode></td>
-            <td><tt>get /</tt></td>
-        </tr>
-        <tr>
-            <td><sourcecode type="http-message">
-CONNECT server.example.com:80 HTTP/1.1
-Host: server.example.com</sourcecode></td>
-            <td><tt>connect /</tt></td>
-        </tr>
-        <tr>
-            <td><sourcecode type="http-message">
-OPTIONS * HTTP/1.1
-Host: server.example.com</sourcecode></td>
-            <td><tt>options *</tt></td>
-        </tr>
-    </tbody>
-</table>
+@authority
+: The authority for a request. ({{content-request-authority}})
+
+@scheme
+: The scheme of the requested URI. ({{content-request-scheme}})
+
+@request-target
+: The request target. ({{content-request-target}})
```

Comment:
I like that you have a shortcut for the whole URI (request-uri) and the  individual components (scheme, authority, path, query-param), but that leaves `@request-target` as a bit of an odd medium.  Perhaps it's time to deprecate it.

Reply:
Perhaps, the OAuth PoP draft didn't have it. I kept it in here because it's relatively easy to get from an HTTP/1.1 message's request line. We'd basically be saying "anything past the authority part of the URL", really.

Reply:
It is less easy to get from HTTP/2 and HTTP/3.  It's somewhat fragile as a consequence of this depending on HTTP/1.1-specific rules for request construction.

Reply:
Good to know about the distinctions. Since each of these identifiers is optional for use we can point out the difficulty in constructing it for HTTP/2 and HTTP/3 in the text, so people working in that space can choose the individual components instead.

Reply:
I added warnings about this identifier in its definition below and will incorporate it in the next version for discussion, if the group chooses to remove it we can do so.